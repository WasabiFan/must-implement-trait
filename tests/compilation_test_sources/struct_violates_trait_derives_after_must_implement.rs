use must_implement_trait::must_implement_trait;
use std::rc::Rc;

fn main() {
    let instance = MySendStruct { value: 7 };
    let _other = instance.clone();
}

#[must_implement_trait(Send)]
#[derive(Clone)]
struct MySendStruct {
    value: Rc<i32>
}