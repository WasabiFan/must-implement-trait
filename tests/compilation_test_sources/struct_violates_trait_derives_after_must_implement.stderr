error[E0277]: `std::rc::Rc<i32>` cannot be sent between threads safely
 --> $DIR/struct_violates_trait_derives_after_must_implement.rs:9:1
  |
9 | #[must_implement_trait(Send)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::rc::Rc<i32>` cannot be sent between threads safely
  |
  = help: within `MySendStruct`, the trait `std::marker::Send` is not implemented for `std::rc::Rc<i32>`
  = note: required because it appears within the type `MySendStruct`
  = help: see issue #48214
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
